interfaces/repo_test.go

```diff
@@ -139,26 +185,26 @@ func (s *RepositorySuite) TestAddPlugClash(c *C) {
 	err = s.testRepo.AddPlug(s.plug)
 	c.Assert(err, ErrorMatches, `cannot add plug, snap "provider" already has plug "plug"`)
 	c.Assert(s.testRepo.AllPlugs(""), HasLen, 1)
-	c.Assert(s.testRepo.Plug(s.plug.Snap, s.plug.Name), DeepEquals, s.plug)
+	c.Assert(s.testRepo.Plug(s.plug.Snap.Name, s.plug.Name), DeepEquals, s.plug)
 }
 
 func (s *RepositorySuite) TestAddPlugFailsWithInvalidSnapName(c *C) {
-	plug := &Plug{
-		Snap:      "bad-snap-",
-		Name:      "name",
-		Interface: "interface",
-	}
+	plug := plugFromYaml(c, "name", []byte(`
+name: bad-snap-
+plugs:
+    name: interface
+`))
 	err := s.testRepo.AddPlug(plug)
 	c.Assert(err, ErrorMatches, `invalid snap name: "bad-snap-"`)
```

Comment:
This is a bogus test now. It's not AddPlug that should complain about the bad snap name, but the yaml parsing logic.